From: Stephane Glondu <steph@glondu.net>
Date: Thu, 26 Dec 2013 21:56:57 +0100
Subject: Fix lost locations in out-of-bounds exceptions

Origin: upstream SVN, r14325
Bug: http://caml.inria.fr/mantis/view.php?id=6233
---
 asmcomp/schedgen.ml | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/asmcomp/schedgen.ml b/asmcomp/schedgen.ml
index a04ab27..e04eacd 100644
--- a/asmcomp/schedgen.ml
+++ b/asmcomp/schedgen.ml
@@ -336,9 +336,8 @@ method private reschedule ready_queue date cont =
             if son.emitted_ancestors = son.ancestors then
               new_queue := son :: !new_queue)
           node.sons;
-        instr_cons_debug node.instr.desc node.instr.arg node.instr.res
-        node.instr.dbg
-          (self#reschedule !new_queue (date + issue_cycles) cont)
+        { node.instr with next =
+            self#reschedule !new_queue (date + issue_cycles) cont }
   end
 
 (* Entry point *)
-- 
