From: Peter Michael Green <plugwash@raspbian.org>
Date: Sat, 7 Dec 2019 05:28:18 +0000
X-Dgit-Generated: 4.08.1-5+rpi1 4372289b19cfc4744b8e5c224da3b2f94ff5285d
Subject: Remove gitattributes that cause problems for dgit.


---

Index: tempdefuzz/.gitattributes
===================================================================
--- tempdefuzz.orig/.gitattributes
+++ tempdefuzz/.gitattributes
@@ -140,7 +140,6 @@ menhir-bench.bash typo.missing-header ty
 *.check-program-output text eol=lf
 *.run text eol=lf
 
-/tools/ci/appveyor/appveyor_build.cmd text eol=crlf
 
 configure.ac text eol=lf
 build-aux/compile text eol=lf
