From: Stefano Zacchiroli <zack@debian.org>
Date: Tue, 19 May 2009 17:24:14 +0200
Subject: Don't use rpath

---
 tools/ocamlmklib.ml | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/tools/ocamlmklib.ml b/tools/ocamlmklib.ml
index 23a273e..a5e0d34 100644
--- a/tools/ocamlmklib.ml
+++ b/tools/ocamlmklib.ml
@@ -40,6 +40,11 @@ and rpath = ref []          (* rpath options *)
 and debug = ref false       (* -g option *)
 and verbose = ref false
 
+(* Debian specific: inhibit rpath *)
+let byteccrpath = ""
+and nativeccrpath = ""
+and mksharedlibrpath = ""
+
 let starts_with s pref =
   String.length s >= String.length pref &&
   String.sub s 0 (String.length pref) = pref
