From: =?utf-8?q?St=C3=A9phane_Glondu?= <glondu@debian.org>
Date: Wed, 12 Jun 2024 13:09:53 +0200
Subject: Fix call to test in configure.ac

Forwarded: https://github.com/ocaml/ocaml/pull/13268
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 467099f..a8e4222 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2690,7 +2690,7 @@ ${mkdll_ldflags}"
   ],[
     mkdll_ldflags='$(OC_DLL_LDFLAGS) $(LDFLAGS)'
     mkdll_ldflags_exp="${oc_dll_ldflags}"
-    AS_IF([test -n ${LDFLAGS}],
+    AS_IF([test -n "${LDFLAGS}"],
       [mkdll_ldflags_exp="$mkdll_ldflags_exp $LDFLAGS"])
     mkexe_ldflags="\$(OC_LDFLAGS) \$(LDFLAGS)"
     mkexe_ldflags_exp="${oc_ldflags} ${LDFLAGS}"
