From: Stephane Glondu <steph@glondu.net>
Date: Mon, 29 Jul 2019 12:49:31 +0200
Subject: Tune testpreempt test to make it work on kfreebsd

---
 testsuite/tests/lib-systhreads/testpreempt.ml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/testsuite/tests/lib-systhreads/testpreempt.ml b/testsuite/tests/lib-systhreads/testpreempt.ml
index 11d6915..f4ef378 100644
--- a/testsuite/tests/lib-systhreads/testpreempt.ml
+++ b/testsuite/tests/lib-systhreads/testpreempt.ml
@@ -19,6 +19,7 @@ let rec generate_list n =
   aux [] n
 
 let rec long_computation time0 =
+  Thread.delay 0.1;
   let long_list = generate_list 100000 in
   let res = List.length (List.rev_map sin long_list) in
   if Sys.time () -. time0 > 2. then
