From: Peter Michael Green <plugwash@raspbian.org>
Subject: Remove gitattributes that cause problems for dgit.

diff --git a/.gitattributes b/.gitattributes
index 0693c28c..87bbecc0 100644
--- a/.gitattributes
+++ b/.gitattributes
@@ -154,7 +154,6 @@ tools/magic                       typo.missing-header
 *.check-program-output text eol=lf
 *.run text eol=lf
 
-/tools/ci/appveyor/appveyor_build.cmd text eol=crlf
 
 Makefile* linguist-language=Makefile
 *.c linguist-language=C
